### DISCUSSION TITLE
Accessory Identify routine?

### DISCUSSION THREAD

**Discussion by aevins** [HomeSpan/HomeSpan#777] [Q&A] [answered]
In HAP, there is an Identify method that can be used to have an accessory identify itself (e.g., blink a lightbulb a couple of times so you know which one you are setting up). As deep as I can search in the source, I cannot find an implementation of this functionality. Is this missing, or something that would be able to be implemented?

[Accessory Identify method - Apple Developer Reference](https://developer.apple.com/documentation/homekit/hmaccessory/1615259-identify)

**Comment by HomeSpan** [comment, marked as answer]
Just checking you were able to resolve this.  It's the Identify Characteristic under the Accessory Information Service that will be triggered with an update when the identification routing is called.  Similar to how you handle the update for any other Characteristic, you add your own code to perform whatever tasks you want when the Identify Characteristic is updated.

Unfortunately, there is no way to trigger this from the Home App itself, so it does not serve much of a purpose.  I think there was a mechanism to do this in much earlier versions of iOS.   Even more ironic is that HomeKit requires this Characteristic be present else it will not pair the device,  even though it's never used.

However, the Eve for HomeKit app does have an explicit button to trigger the ID routine and it works well to help identify a device.

**Reply by aevins** [reply]
Perfect, thank you! Mixiaoxiao/Arduino-HomeKit-ESP8266 contained an explicit way to define a callback for this, but HomeSpan implementation makes more sense.

On pairing a new accessory is the only time the Home app provides the option to Identify, and I'm definitely interested in having my project react to this.

I wonder if this may be something to explore and write into the examples, since it really only gets used at one point in time, but is really cool to implement if you want a feature-complete accessory. Thanks again!
